linetools/isgm/utils.py

```diff
@@ -362,3 +366,138 @@ def synthesize_components(components, zcomp=None, vbuff=0*u.km/u.s):
 
     # Return
     return synth_comp
+
+
+def overlapping_chunks(chunk1, chunk2):
+    """True if there is overlap between chunks
+    `chunk1` and `chunk2`. Otherwise False. Chunks are
+    assumed to represent continuous coverage, so the only
+    information that matters are the minimum and maximum
+    values of a given chunk.
+
+    Parameters
+    ----------
+    chunk1 : tuple, list, 1-d np.array, Quantity
+        A given chunk, assumed to represent a contiguous region
```

Comment:
Does it need to be iterable or can it be a single number (as you suggest with Quantity)?

If iterable, I'd check it with astropy.utils.misc.isiterable